modules/basis/basis-datasource/tomcat-listener/src/main/java/io/dirigible/tomcat/listener/NamingResourcesListener.java

```diff
@@ -0,0 +1,156 @@
+package io.dirigible.tomcat.listener;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.catalina.Context;
+import org.apache.catalina.Lifecycle;
+import org.apache.catalina.LifecycleEvent;
+import org.apache.catalina.LifecycleListener;
+import org.apache.catalina.deploy.NamingResourcesImpl;
+import org.apache.naming.ContextAccessController;
+import org.apache.tomcat.util.descriptor.web.ContextResource;
+import org.apache.tomcat.util.descriptor.web.ResourceBase;
+
+public class NamingResourcesListener implements LifecycleListener, PropertyChangeListener {
+
+  private static final Logger LOGGER = Logger.getLogger(NamingResourcesListener.class.getName());
+
+  private static final String RESOURCE_STR = "resource";
+
+  private static final String RESOURCE_ENV_REF_STR = "resourceEnvRef";
+
+  static final String FACTORY_CLASS_NAME = "io.dirigible.listener.DelegatingObjectFactory";
+
+  @Deprecated
+  static final String PERSISTENCE_FACTORY_CLASS_NAME = "com.sap.jpaas.service.persistence.core.JNDIDataSourceFactory";
+
+  static final String FACTORY_PROPERTY = "factory";
+
+  private Object container = null;
+
+  private NamingResourcesImpl namingResources = null;
+
+  private boolean initialized = false;
+
+  private static final String name = "/";
+
+  private String factoryClassName;
+
+  static final String RESOURCE_REF_NAME = "ResourceRefName";
+
+  private static final String DATA_SOURCE = "DataSource";
+
+  @Override
+  public void propertyChange(PropertyChangeEvent event) {
+    if (!this.initialized) {
+      return;
+    }
+
+    final Object source = event.getSource();
+    if (source == this.namingResources) {
```

Comment:
Shouldn't the check be with `.equals()`?

Reply:
Is there a difference between '==' and equals() ?